src/Common/Session/SessionUtil.php

```diff
@@ -75,20 +75,22 @@ public static function coreSessionStart($web_root, $read_only = true): void
     {
         // Note there is no system logger here since that class does not
         //  yet exist in this context.
-        session_start([
-            'read_and_close' => $read_only,
-            'cookie_samesite' => self::$use_cookie_samesite,
-            'cookie_secure' => self::$use_cookie_secure,
-            'name' => 'OpenEMR',
-            'cookie_httponly' => false,
-            'cookie_path' => ((!empty($web_root)) ? $web_root . '/' : '/'),
-            'gc_maxlifetime' => self::$gc_maxlifetime,
-            'sid_bits_per_character' => self::$sid_bits_per_character,
-            'sid_length' => self::$sid_length,
-            'use_strict_mode' => self::$use_strict_mode,
-            'use_cookies' => self::$use_cookies,
-            'use_only_cookies' => self::$use_only_cookies,
-        ]);
+        if (session_status() === PHP_SESSION_NONE) {
```

Comment:
This kind of thing makes me nervous (i think then any session that is not core could circumvent core). What is the use case for this?

Reply:
I suppose one could start a session and prevent starting a named session. Regardless if you can create a session, you can create one using the OpenEMR name. Still whichever session is active, all that will happen is a noisy PHP notice.
The issue stems from our setSession() method does a session start  to allow write mode since we default to read only. https://github.com/openemr/openemr/blob/61e4f5bc1d0d5c12661ad2d46b1d398426d138ca/src/Common/Session/SessionUtil.php#L94-L97 

Now we assume that since we set the read_and_close option that the session will be closed when attempting a session write and the session will be reopened in a write mode. Not always true plus if indeed the session is still open it may or may not mean the session file wasn't written from a previous write open with a session_write_close().
This may explain some of our random session timeout issues.

My point is we need to ensure our session state when possible and if a session is open when calling the core session start then we possibly missed a session save. This started with me getting rid of annoying php warning which in my case is caused by doing a session write from module though I still use the session util i.e
```
public function setSession($key, $value): static
    {
        // ensure write is allowed by using utility.
        SessionUtil::setSession($key, $value);
        return $this;
    }
```
So unknow why the wrong session state.

I think a better solution is
```
if (session_status() === PHP_SESSION_ACTIVE) {
            session_write_close();
        }
```

Furthermore it is time we implement our own session manager because we are beyond what the basic PHP handler is designed for!